scripts-base/src/componentTypes.ts

```diff
@@ -0,0 +1,76 @@
+export type ComponentEl = HTMLElement | SVGElement | Window
+
+export type EventMapByElement<E> = E extends Window
+	? WindowEventMap
+	: E extends SVGElement
+	? SVGElementEventMap
+	: E extends HTMLBodyElement
+	? HTMLBodyElementEventMap
+	: E extends HTMLVideoElement
+	? HTMLVideoElementEventMap
+	: E extends HTMLAudioElement
+	? HTMLMediaElementEventMap
+	: E extends HTMLElement
+	? HTMLElementEventMap
+	: never
```

Comment:
Jde tento zápis zpřehlednit závorkami a odsazením, nebo to TS vyžaduje takto?

Reply:
Fixed.

Reply:
😂 😂
Jako vážně je tohle lepší?

Původní Jirkův zápis není nikterak neobvyklý.  Nevytváří příliš hluboké odsazení, subjektivně se rozhodně čte lépe.

![image](https://user-images.githubusercontent.com/1045362/60592325-ae883100-9da0-11e9-9999-40cccf64bb56.png)

Reply:
Je to celé psané na `indent_size = 2`, tak je to pak docela v pohodě. :man_shrugging: